hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/StorageContainerManager.java

```diff
@@ -879,14 +880,20 @@ certificateStore, new DefaultCAProfile(),
     SecretKeyManager secretKeyManager = secretKeyManagerService != null ?
         secretKeyManagerService.getSecretKeyManager() : null;
 
+    X509Certificate rootCaCert = scmCertificateClient == null ? null :
+        scmCertificateClient.getRootCACertificate() != null ?
+            scmCertificateClient.getRootCACertificate() :
+            scmCertificateClient.getCACertificate();
+    ArrayList<X509Certificate> rootCaList = new ArrayList<>();
+    rootCaList.add(rootCaCert);
```

Comment:
If `rootCaCert` is null based on above condition, in that case we should not add into `rootCaList`.

Reply:
Thanks, nice catch